ReactNative/ReactTwitter/core/views/splash-screen.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import {
+  StyleSheet,
+  View,
+  Text,
+  Navigator
+} from 'react-native'
+var AuthenticationService = require('../service/authentication-service.js')
+
+const SPLASH_DURATION_MS = 1000
+
+var SplashScreenView = React.createClass({
+  propTypes: {
+    navigator: React.PropTypes.instanceOf(Navigator).isRequired
+  },
+
+  getInitialState () {
+    return {
+      authenticationService: new AuthenticationService()
+    }
+  },
+
+  componentDidMount () {
+    let authService = this.state.authenticationService
+
+    authService.loadDataFromDisk()
+      .then(() => {
+        return new Promise((resolve, reject) => {
+          setTimeout(resolve, SPLASH_DURATION_MS)
+        })
+      })
+      .then(() => {
+        if (authService.isUserLoggedIn()) {
+          this._pushTweetsList()
+        } else {
+          this._pushLogin()
+        }
+      })
```

Comment:
I know it's never the case, but should we also put here a catch statement for possible errors?